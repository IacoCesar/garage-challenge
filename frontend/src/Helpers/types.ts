export type TNavigation = {
  navigate: (route: string, params?: object) => void
}
