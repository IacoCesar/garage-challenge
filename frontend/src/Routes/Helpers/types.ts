export type TPage = {
  [key: string]: string
}

export interface IProps {
  isHomeScreen: boolean
  title: string
}
