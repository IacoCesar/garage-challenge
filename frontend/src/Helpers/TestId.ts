import { Platform } from 'react-native'

interface ITestID {
  testID: string
  accessibilityLabel?: string
  accessible?: boolean
}

type componentType =
  | 'TouchableOpacity'
  | 'View'
  | 'SafeAreaView'
  | 'List'
  | 'Modal'
  | 'Text'
  | 'Image'
  | 'ActivityIndicator'

const testID = (testID: string, accessibilityLabel?: string): ITestID =>
  Platform.OS === 'android'
    ? {
        testID,
        accessibilityLabel,
        accessible: true
      }
    : {
        testID,
        accessible: false
      }

export const assignTestId = (
  componentType: componentType,
  componentId: string,
  componentText?: string
): ITestID => {
  const id = `${componentType}_${componentId}`

  return testID(id, componentText)
}

export const mockTestId = (
  componentType: componentType,
  componentId: string,
  componentText?: string
): ITestID => {
  return assignTestId(componentType, componentId, componentText)
}
