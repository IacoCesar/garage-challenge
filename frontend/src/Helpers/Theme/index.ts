import { DefaultTheme } from 'styled-components/native'

export const commonThemes = {
  colors: {
    bgWhite: '#EFEDEE',
    white: '#FFFF',
    black: '#000000',
    cardColor: '#ffffff',
    starColor: '#F9E26B',
    divisorColor: '#9F9F9F'
  },
  spacing: {
    global: {
      horizontal: 16
    }
  }
}

export const theme: DefaultTheme = {
  standard: {
    darkMode: {
      icon: commonThemes.colors.white,
      divisor: commonThemes.colors.bgWhite,
      shadow: commonThemes.colors.white,
      card: commonThemes.colors.divisorColor,
      cardTextColor: commonThemes.colors.black,
      background: commonThemes.colors.black,
      textColor: commonThemes.colors.white,
      ...commonThemes
    },
    default: {
      icon: commonThemes.colors.black,
      divisor: commonThemes.colors.divisorColor,
      shadow: commonThemes.colors.black,
      card: commonThemes.colors.cardColor,
      cardTextColor: commonThemes.colors.black,
      background: commonThemes.colors.white,
      textColor: commonThemes.colors.black,
      ...commonThemes
    }
  },
  ...commonThemes
}
