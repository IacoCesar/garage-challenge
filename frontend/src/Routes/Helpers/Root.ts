import RootPdp from '@/Modules/ProductDetailPage'
import RootPlp from '@/Modules/ProductListPage'

export const root = {
  ProductDetailPage: RootPdp.Routes.ProductDetailPage,
  ProductListPage: RootPlp.Routes.ProductListPage
}
