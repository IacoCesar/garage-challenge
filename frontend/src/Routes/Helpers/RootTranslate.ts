import { TPage } from './types'

const pageParse: TPage = {
  ProductListPage: 'Garage',
  ProductDetailPage: 'Detail'
}

export default {
  pageParse
}
