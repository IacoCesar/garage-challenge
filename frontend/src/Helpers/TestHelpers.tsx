import React from 'react'
import { SafeAreaProvider } from 'react-native-safe-area-context'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '@/Helpers/Theme'
import configureStore from 'redux-mock-store' //ES6 modules

import { render } from '@testing-library/react-native'

const mockStore = configureStore([])

const renderProviders = ({ children }: any): React.ReactElement => {
  const store = mockStore({
    state: { themeData: { theme: 'default' } }
  })

  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <SafeAreaProvider>{children}</SafeAreaProvider>
      </ThemeProvider>
    </Provider>
  )
}

const customRender = (ui: React.ReactElement, options?: object) =>
  render(ui, {
    wrapper: renderProviders,
    ...options
  })

export * from '@testing-library/react-native'

export { customRender as render }
