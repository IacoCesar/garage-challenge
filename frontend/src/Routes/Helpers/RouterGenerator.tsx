import React from 'react'
import Header from '@/Components/Header'
import RootTranslate from './RootTranslate'
import { IProps } from './types'
import { useDispatch } from 'react-redux'
import { changeTheme } from '@/Redux/Theme'

export const RouteGenerator = ({ isHomeScreen, title }: IProps) => {
  const dispatch = useDispatch()

  const callHandleBack = (navigation: any) => {
    navigation.goBack()
  }

  const callHandleChangeVisualMode = (themeSelector: string) => {
    dispatch(changeTheme(themeSelector === 'darkMode' ? 'default' : 'darkMode'))
  }

  return {
    header: ({ navigation }: any) => {
      return (
        <Header
          callHandleChangeVisualMode={callHandleChangeVisualMode}
          callHandleBack={() => callHandleBack(navigation)}
          isHomeScreen={isHomeScreen}
          title={RootTranslate.pageParse[title] || title}
        />
      )
    }
  }
}
