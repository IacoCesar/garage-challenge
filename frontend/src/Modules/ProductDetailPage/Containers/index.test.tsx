import React from 'react'
import DetaiProductDetailPagel from '.'
import { render } from '@/Helpers/TestHelpers'

describe('ProductDetailPage', () => {
  it('should render snapshot', async () => {
    const wrapper = render(<DetaiProductDetailPagel />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
