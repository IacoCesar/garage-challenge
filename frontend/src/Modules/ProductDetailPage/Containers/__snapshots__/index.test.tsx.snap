// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailPage should render snapshot 1`] = `
<RCTScrollView
  ListHeaderComponent={
    <React.Fragment>
      <Styled(View)
        accessible={false}
        testID="View_Detail_Container_Content"
        themeSelector="default"
      >
        <SpacerView
          max={16}
          min={16}
          testID="Detail_Container_Content_Spacer"
        />
        <Styled(Image)
          accessible={false}
          resizeMode="stretch"
          source={
            Object {
              "uri": "http://localhost:3000/assets/vehicle_image-155fa5ae-00a0-11ea-84e3-4371baa68b28.jpg",
            }
          }
          testID="Image_Detail_Container_Content_Image"
          themeSelector="default"
        />
        <SpacerView
          max={20}
          min={20}
          testID="Detail_Container_Content_Spacer_1"
        />
        <Styled(View)
          accessible={false}
          testID="View_Detail_Container_Content_ContentInfo"
        >
          <Styled(View)
            accessible={false}
            testID="View_Detail_Container_Content_ContentInfo_ContentTextIdentifier"
          >
            <Component
              fontSize={12}
              label="ID: 155fa5ae-00a0-11ea-84e3-4371baa68b28"
              testID="Detail_Container_Content_ContentInfo_ContentTextIdentifier_Text"
            />
          </Styled(View)>
          <Styled(View)
            accessible={false}
            testID="View_Detail_Container_Content_ContentInfo_ContentHr"
            themeSelector="default"
          />
          <SpacerView
            max={5}
            min={5}
            testID="Detail_Container_Content_ContentInfo_Spacer"
          />
          <Component
            fontSize={20}
            label="Make: Audi"
            testID="Detail_Container_Content_ContentInfo_Text"
          />
          <SpacerView
            max={5}
            min={5}
            testID="Detail_Container_Content_ContentInfo_Spacer_1"
          />
          <Component
            fontSize={20}
            label="Model: RS7 4.0"
            testID="Detail_Container_Content_ContentInfo_Text_1"
          />
          <SpacerView
            max={5}
            min={5}
            testID="Detail_Container_Content_ContentInfo_Spacer_2"
          />
          <Component
            fontSize={20}
            label="Year: 2015"
            testID="Detail_Container_Content_ContentInfo_Text_2"
          />
        </Styled(View)>
      </Styled(View)>
    </React.Fragment>
  }
  accessible={false}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="List_Detail_Container"
  themeSelector="default"
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        accessible={false}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFF",
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
        testID="View_Detail_Container_Content"
        themeSelector="default"
      >
        <View
          accessible={false}
          max={16}
          min={16}
          style={
            Array [
              Object {
                "flexGrow": 1,
                "maxHeight": 16,
                "minHeight": 16,
              },
            ]
          }
          testID="View_Detail_Container_Content_Spacer"
        />
        <Image
          accessible={false}
          resizeMode="stretch"
          source={
            Object {
              "uri": "http://localhost:3000/assets/vehicle_image-155fa5ae-00a0-11ea-84e3-4371baa68b28.jpg",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "#9F9F9F",
                "display": "flex",
                "height": 421.875,
                "width": "100%",
              },
            ]
          }
          testID="Image_Detail_Container_Content_Image"
          themeSelector="default"
        />
        <View
          accessible={false}
          max={20}
          min={20}
          style={
            Array [
              Object {
                "flexGrow": 1,
                "maxHeight": 20,
                "minHeight": 20,
              },
            ]
          }
          testID="View_Detail_Container_Content_Spacer_1"
        />
        <View
          accessible={false}
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "column",
                "paddingHorizontal": 16,
              },
            ]
          }
          testID="View_Detail_Container_Content_ContentInfo"
        >
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "width": "100%",
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_ContentTextIdentifier"
          >
            <View
              accessible={false}
              style={
                Array [
                  Object {
                    "display": "flex",
                  },
                ]
              }
              testID="View_Detail_Container_Content_ContentInfo_ContentTextIdentifier_Text"
            >
              <Text
                accessible={false}
                color="#000000"
                fontSize={12}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="Text_Detail_Container_Content_ContentInfo_ContentTextIdentifier_Text_Content_Text"
              >
                ID: 155fa5ae-00a0-11ea-84e3-4371baa68b28
              </Text>
            </View>
          </View>
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "borderColor": "#9F9F9F",
                  "borderWidth": 2,
                  "display": "flex",
                  "width": "100%",
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_ContentHr"
            themeSelector="default"
          />
          <View
            accessible={false}
            max={5}
            min={5}
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "maxHeight": 5,
                  "minHeight": 5,
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_Spacer"
          />
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "display": "flex",
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_Text"
          >
            <Text
              accessible={false}
              color="#000000"
              fontSize={20}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 20,
                  },
                ]
              }
              testID="Text_Detail_Container_Content_ContentInfo_Text_Content_Text"
            >
              Make: Audi
            </Text>
          </View>
          <View
            accessible={false}
            max={5}
            min={5}
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "maxHeight": 5,
                  "minHeight": 5,
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_Spacer_1"
          />
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "display": "flex",
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_Text_1"
          >
            <Text
              accessible={false}
              color="#000000"
              fontSize={20}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 20,
                  },
                ]
              }
              testID="Text_Detail_Container_Content_ContentInfo_Text_1_Content_Text"
            >
              Model: RS7 4.0
            </Text>
          </View>
          <View
            accessible={false}
            max={5}
            min={5}
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "maxHeight": 5,
                  "minHeight": 5,
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_Spacer_2"
          />
          <View
            accessible={false}
            style={
              Array [
                Object {
                  "display": "flex",
                },
              ]
            }
            testID="View_Detail_Container_Content_ContentInfo_Text_2"
          >
            <Text
              accessible={false}
              color="#000000"
              fontSize={20}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 20,
                  },
                ]
              }
              testID="Text_Detail_Container_Content_ContentInfo_Text_2_Content_Text"
            >
              Year: 2015
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
